types/basetypes/int32_type_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package basetypes
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+)
+
+func TestInt32TypeValueFromTerraform(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		input       tftypes.Value
+		expectation attr.Value
+		expectedErr string
+	}
+	tests := map[string]testCase{
+		"value": {
+			input:       tftypes.NewValue(tftypes.Number, 123),
+			expectation: NewInt32Value(123),
+		},
+		"unknown": {
+			input:       tftypes.NewValue(tftypes.Number, tftypes.UnknownValue),
+			expectation: NewInt32Unknown(),
+		},
+		"null": {
+			input:       tftypes.NewValue(tftypes.Number, nil),
+			expectation: NewInt32Null(),
+		},
+		"wrongType": {
+			input:       tftypes.NewValue(tftypes.String, "oops"),
+			expectedErr: "can't unmarshal tftypes.String into *big.Float, expected *big.Float",
+		},
```

Comment:
Since we have some logic around the bounds/validation of the value, we should add some additional tests for:
- `math.Int32Min` success
- `math.Int32Max` success
- `math.Int32Min-1` failure
- `math.Int32Max+1` failure
- Maybe a test that something like `tftypes.NewValue(tftypes.Number, 123.1)` returns an error